internal/resources/slo/resource_slo.go

```diff
@@ -0,0 +1,477 @@
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"key": &schema.Schema{
+													Type:     schema.TypeString,
+													Required: true,
+												},
+												"value": &schema.Schema{
+													Type:     schema.TypeString,
+													Required: true,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceSloCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	slo, err := packSloResource(d)
+	if err != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Unable to Pack SLO",
+			Detail:   err.Error(),
+		})
+		return diags
+	}
+
+	client := m.(*common.Client).GrafanaAPI
+	response, err := client.CreateSlo(slo)
+
+	if err != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Unable to create SLO - API",
+			Detail:   fmt.Sprintf("API Error Message:%s", err.Error()),
+		})
+		return diags
+	}
+
+	d.SetId(response.UUID)
+	resourceSloRead(ctx, d, m)
+
+	return diags
+}
+
+// resourceSloRead - sends a GET Request to the single SLO Endpoint
+func resourceSloRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	sloID := d.Id()
+
+	client := m.(*common.Client).GrafanaAPI
+	slo, err := client.GetSlo(sloID)
+
+	if err != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  fmt.Sprintf("Unable to Fetch Slo with ID: %s", sloID),
+			Detail:   fmt.Sprintf("API Error Message:%s", err.Error()),
+		})
+		return diags
+	}
+
+	setTerraformState(d, slo)
+
+	return diags
+}
+
+func resourceSloUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	sloID := d.Id()
+
+	if d.HasChange("name") || d.HasChange("description") || d.HasChange("query") || d.HasChange("labels") || d.HasChange("objectives") || d.HasChange("alerting") {
+		slo, err := packSloResource(d)
+		if err != nil {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Unable to Pack SLO",
+				Detail:   err.Error(),
+			})
+			return diags
+		}
+
+		client := m.(*common.Client).GrafanaAPI
+
+		err = client.UpdateSlo(sloID, slo)
+		if err != nil {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  fmt.Sprintf("Unable to Update Slo with ID: %s", sloID),
+				Detail:   fmt.Sprintf("API Error Message:%s", err.Error()),
+			})
+			return diags
+		}
+	}
+
+	return resourceSloRead(ctx, d, m)
+}
+
+func resourceSloDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	sloID := d.Id()
+
+	client := m.(*common.Client).GrafanaAPI
+	client.DeleteSlo(sloID)
+
+	d.SetId("")
+
+	return diags
```

Comment:
Error is being ignored here and Terraform deletes the ID for you. So this func could be simplified
```suggestion
	sloID := d.Id()
	client := m.(*common.Client).GrafanaAPI
	return diag.FromError(client.DeleteSlo(sloID))
```